schema/types/semver.go

```diff
@@ -30,6 +30,12 @@ func NewSemVer(s string) (*SemVer, error) {
 	return &v, nil
 }
 
+func (sv SemVer) LessThan(versionB SemVer) bool {
```

Comment:
Perhaps the default would check against something like `LessThanMajor()`, and have an option for strict checking and have this function called `LessThanExact()` or similar.